### ISSUE TITLE
[consistent-return/default-case] If enum is exhausted in switch, default-case should not be triggered.

### ISSUE THREAD

**Issue by vipcxj** [typescript-eslint/typescript-eslint#2284]
**Repro**

```JSON
{
  "rules": {}
}
```

```TS
enum Enum {
  A, B, C
}
// trigger consistent-return
function Text(arg: Enum): string {
  // trigger default-case
  switch(arg) {
    case Enum.A: return 'a';
    case Enum.B: return 'b';
    case Enum.C: return 'c';
  }
}
```

**Expected Result**
no error

**Actual Result**
trigger consistent-return and default-case

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.6.0` |
| `@typescript-eslint/parser`        | `3.6.0` |
| `TypeScript`                       | `3.9.6` |
| `ESLint`                           | `7.4.0` |
| `node`                             | `14.4.0` |
| `npm`                              | `6.14.5` |

**Comment by bradzacher** [comment]
Instead of `default-case`, consider using our `switch-exhaustiveness-check` rule.

I would consider not using `consistent-return` at all - typescript will do these checks for you (`Not all code paths return a value. (7030)`).

**Comment by vipcxj** [comment]
Is `switch-exhaustiveness-check` in the recommand list? And can it completely replace the functional of `default-case`?

**Comment by bradzacher** [comment]
> Is switch-exhaustiveness-check in the recommand list?

No, but neither is `default-case`.

> can it completely replace the functional of default-case?

Almost.
It only works for union types (`type T = 1 | 2 | 3; type U = 'a' | 'b' | 'c'`) and enums.
Check out the docs for the rule https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/switch-exhaustiveness-check.md

**Comment by vipcxj** [comment]
Is there a way to let `default-case` not complain when `switch-exhaustiveness-check` make sure it fixes everything？

**Comment by bradzacher** [comment]
no. each lint rule is independent.
